### PR TITLE
Podman AI Lab to RHOAI

### PR DESCRIPTION
**Pull request by jhurlocker**
This is a tutorial to showing how a developer can take the chatbot recipe in Podman AI Lab and modify it with Langchain to create a simple RAG chatbot. Steps to deploy Elasticsearch vector DB, ingest data, and deploy an LLM on OpenShift AI are included.

### CODE REVIEW

File: docs/demos/podman-ai-lab-to-rhoai/podman-ai-lab-to-rhoai.md [hunk 1]
```diff
@@ -0,0 +1,447 @@
...
+Make sure to select _TheBloke/Mistral-7B-Instruct-v0.2-GGUF_ as your model and then click **_Start AI App_** button.
+![Start Chatbot](img/start_chatbot_recipe.png)
+
+6. After the chatbot has started open it up to test it out.
+![Chatbot UI](img/chatbot_recipe_UI.png)
+
+7. At the bottom of the AI App Details section you'll see a **_Open in VSCode_** button. Clicking on that will open all of the code that is running your chatbot. Later we'll modify that code to connect langchain, TheBloke/Mistral-7B-Instruct-v0.2-GGUF model, and the Elasticsearch Vector Database. 
+![Podman AI Chatbot Recipe](img/podman_AI_chatbot.png)
+
+## Deploying OpenShift AI
```
Comment (guimou) [inline docs/demos/podman-ai-lab-to-rhoai/podman-ai-lab-to-rhoai.md:71]: This section should be optional, or point to OpenShift AI installation instructions. Most people won't start from a vanilla OpenShift cluster.

File: docs/demos/podman-ai-lab-to-rhoai/podman-ai-lab-to-rhoai.md [hunk 2]
```diff
@@ -0,0 +1,447 @@
...
+    ```
+
+    ![OpenShift AI Operator Status](img/oc_rhoai_operator_status.png)
+
+    Once the pod has a Running status and is ready you can run the below command.
+    + ```
+    oc apply -k ./components/openshift-ai/instance/overlays/fast
+    ```
+
+## Deploy Elasticsearch Vector DB
```
Comment (guimou) [inline docs/demos/podman-ai-lab-to-rhoai/podman-ai-lab-to-rhoai.md:98]: Same as before, I would set this in its own repo/doc or mark it optional.

File: docs/demos/podman-ai-lab-to-rhoai/podman-ai-lab-to-rhoai.md [hunk 3]
```diff
@@ -0,0 +1,447 @@
...
+
+    This cell will take a while to run. Good time for a coffee break.
+
+    ![Notebook Execution - Second Section](img/ingest_data_cell.png)
+
+10. After all of the data is stored into our vector database we can directly query it. Run the last 2 cells to make sure the data was stored successfully and we get back the results we expect.
+
+    ![Notebook Execution - Third Section](img/query_data_cell.png)
+
+## Deploy s3 Storage (Minio)
```
Comment (guimou) [inline docs/demos/podman-ai-lab-to-rhoai/podman-ai-lab-to-rhoai.md:192]: There is already a Minio deployment recipe on the site, maybe you could point to it.
This should also be optional, if people don't already have it or other S3 solution deployed.

File: docs/demos/podman-ai-lab-to-rhoai/podman-ai-lab-to-rhoai.md [hunk 4]
```diff
@@ -0,0 +1,447 @@
...
+
+    In Minio click on the **Upload File** button and select the model file under the *hf.TheBloke.mistral-7b-instruct-v0.2.Q4_K_M* directory.
+
+    ![Model upload](img/minio_model_upload.png)
+
+6. If the model is uploaded successfully you should see the below screen.
+
+    ![Model upload success](img/minio_model_upload_success.png)
+
+## Create Custom Model Serving Runtime
```
Comment (guimou) [inline docs/demos/podman-ai-lab-to-rhoai/podman-ai-lab-to-rhoai.md:232]: This is part of RHOAI installation, refer to the doc.

File: docs/demos/podman-ai-lab-to-rhoai/podman-ai-lab-to-rhoai.md [hunk 5]
```diff
@@ -0,0 +1,447 @@
...
+
+        ![Service Mesh Instance](img/servicemesh_instance.png)
+
+2. Go to the OpenShift AI dashboard and expand **Settings** and select **Serving Runtimes**. You should now see that Single-model serving enabled at the top of the page.
+
+    **NOTE:** *You might need to refresh the page and it could take a few minutes for the changes to be applied.*
+
+    ![Single serving enabled](img/rhoai_single_serving_enabled.png)
+
+### Add a Custom Serving Runtime
```
Comment (guimou) [inline docs/demos/podman-ai-lab-to-rhoai/podman-ai-lab-to-rhoai.md:301]: Add a note that we are using a GGUF model just for the sake of the demo.
The whole purpose of RHOAI and the transition from Podman AI to it is to be able to "scale" and use real models (vs quantized) on a supported environment. If we only reuse the same quantized model here there is not much point in using RHOAI.

File: docs/demos/podman-ai-lab-to-rhoai/podman-ai-lab-to-rhoai.md [hunk 6]
```diff
@@ -0,0 +1,447 @@
...
+
+    ![Deployed Model](img/rhoai_deployed_model.png)
+
+7. Test your model to make sure you can send in a request and get a response. You can use the client code that is provided by the model service in Podman AI Lab. 
+
+      Make sure to update the URL in the cURL command to the Inference endpoint on OpenShift AI.
+
+
+      ```
+      curl -k --location 'https://YOUR-OPENSHIFT-AI-INFERENCE-ENDPOINT/v1/chat/completions' --header 'Content-Type: application/json' --data '{
```
Comment (guimou) [inline docs/demos/podman-ai-lab-to-rhoai/podman-ai-lab-to-rhoai.md:353]: Why use `-k` if you went through all the process of certificates imports?

**Review by guimou — COMMENTED** [review]
All the comments are basically around the components that may have been previously installed: RHOAI, Minio, Elastic,...
Marking their installation sections as optional, or pointing to the official doc when it exists (requested by the RHOAI doc team) will help focus on the solution and the architecture.
But apart from that it's a nice example on how PodmanAI and RHOAI can be complementary.

File: docs/demos/podman-ai-lab-to-rhoai/podman-ai-lab-to-rhoai.md (hunk 1 above)
Comment (jhurlocker) [reply docs/demos/podman-ai-lab-to-rhoai/podman-ai-lab-to-rhoai.md]: Made the section optional, included the installation instructions in the product documentation, and made the automated installation collapsible.

File: docs/demos/podman-ai-lab-to-rhoai/podman-ai-lab-to-rhoai.md (hunk 2 above)
Comment (jhurlocker) [reply docs/demos/podman-ai-lab-to-rhoai/podman-ai-lab-to-rhoai.md:106]: Made the section optional, included the installation instructions in the operator documentation, and made the automated installation collapsible.

File: docs/demos/podman-ai-lab-to-rhoai/podman-ai-lab-to-rhoai.md (hunk 3 above)
Comment (jhurlocker) [reply docs/demos/podman-ai-lab-to-rhoai/podman-ai-lab-to-rhoai.md:209]: Made the section optional, included the link to the Mino installation instructions on ai-on-openshift site.

File: docs/demos/podman-ai-lab-to-rhoai/podman-ai-lab-to-rhoai.md (hunk 4 above)
Comment (jhurlocker) [reply docs/demos/podman-ai-lab-to-rhoai/podman-ai-lab-to-rhoai.md:240]: Included the installation instructions in the product documentation and made the automated installation collapsible.

File: docs/demos/podman-ai-lab-to-rhoai/podman-ai-lab-to-rhoai.md (hunk 5 above)
Comment (jhurlocker) [reply docs/demos/podman-ai-lab-to-rhoai/podman-ai-lab-to-rhoai.md:317]: Added "We will continue to use the GGUF version of the model to be able to deploy this model without the need for a hardware accelerator (e.g. GPU). OpenShift AI contains a scalable model serving platform to accommodate deploying multiple full sized LLMs."

File: docs/demos/podman-ai-lab-to-rhoai/podman-ai-lab-to-rhoai.md (hunk 6 above)
Comment (jhurlocker) [reply docs/demos/podman-ai-lab-to-rhoai/podman-ai-lab-to-rhoai.md]: Removed the -k. It was left in by accident.

**Pull request merged**